### ISSUE TITLE
Any way to detect current langage in utils ?

### ISSUE THREAD

**Issue by pyladune** [Gizra/elm-i18n-example#1]
This implementation is pretty nice but would there be a as nice way to detect the language selected by user inside the utils functions ?
The point is to be able to use translate function in submodule without passing the main model each time just for that

**Comment by amitaibu** [comment]
You don't need to pass all the model, just the language and the string ID.

**Comment by pyladune** [comment]
Yes, but how to access to currentlanguage directly from utils.elm
instead to pass it at each call of translate ? that would make the whole
thing easier i think

**Comment by amitaibu** [comment]
You can't,  because in Elm you always pass arguments around.  No global
state.

**Comment by simonewebdesign** [comment]
One way to just pass the string ID to the translate function would be to curry it beforehand, and then you can pass the curried translate function down to the submodule where you need it.

**Comment by pyladune** [comment]
@simonewebdesign could you explain more or illustrate, seems interresting but don't get it at all :-)

**Comment by simonewebdesign** [comment]
@pyladune The translate function has this signature:

``` elm
translate : Language -> TranslationId -> String
```

Curried:

``` elm
curriedTranslate : TranslationId -> String
curriedTranslate trans =
    translate currentLanguage trans


currentLanguage : Language
currentLanguage =
    Japanese
```

At this point you can just use it anywhere without carrying the currentLanguage about :)
